addons/mail/models/discuss/ir_attachment.py

```diff
@@ -27,3 +27,19 @@ def _get_upload_env(self, request, thread_model, thread_id):
                 .env
             )
         return super()._get_upload_env(request, thread_model, thread_id)
+
+    def _prepare_attachment_format(self, attachment):
+        attachment_format = super()._prepare_attachment_format(attachment)
+        voice = self.env["discuss.attachment.metadata"].sudo().search([("attachment_id", "=", attachment.id)])
+        if voice:
+            attachment_format['duration'] = voice.duration
+        return attachment_format
+
+    def create_uploaded_attachment(self, vals, **kwargs):
```

Comment:
`create_uploaded_attachment` is already defined in `mail`. Overriding the method in a subfolder `discuss/` only brings confusion.

Reply:
This will probably be moved to a new a discuss module in the future. This is why you have this override.

Reply:
"will probably in the future" does not give me a lot of confidance. Create the override when you actually makes the move, not now. If you really want to make a subfolder (which I find very confusing), at least make the code in it really independant.

Reply:
I'm sorry, the "probably" is misleading. The current plan is to split in subfolder and then move to a new module. For want I know this plan is still in motion while some of us are creating new feature. We try to not create additional work for the people working on the split.